Poker.BE/Poker.BE.Domain/Game/Turn.cs

```diff
@@ -12,58 +12,34 @@ public class Turn
     {
         #region Fields
         private Player currentPlayer;
-        private Pot currentPot;
         #endregion
 
         #region Properties
-        public Player CurrentPlayer { get { return currentPlayer; } set { currentPlayer = value; } }
-        public Pot CurrentPot { get { return currentPot; } set { currentPot = value; } }
+        /// <summary>
+        /// The set of this property raising an event
+        /// </summary>
+        public Player CurrentPlayer
+        {
+            get
+            {
+                return currentPlayer;
+            }
+            set
+            {
+                currentPlayer = value;
+                currentPlayer.OnTurnChanged(EventArgs.Empty);
+            }
+        }
         #endregion
 
         #region Constructors
-        public Turn(Player player, Pot pot)
+        public Turn(Player player)
         {
             this.CurrentPlayer = player;
-            this.currentPot = pot;
         }
         #endregion
 
         #region Methods
-        public void Check()
-        {
-            ///  Do noting?
-        }
-
-        public void Call(int amount)
-        {
-            if (amount <= 0)
-                throw new IOException("negative call :(  Somthing isn't right...");
-            this.currentPlayer.SubstractMoney(amount);
-        }
-
-        public void Fold()
-        {
-            this.currentPlayer.CurrentState = Player.State.ActiveFolded;
-        }
-
-        public void Bet(int amount)
-        {
-            if (amount <= 0)
-                throw new IOException("Can't bet a negetive amount");
-            this.currentPlayer.SubstractMoney(amount);
-        }
-
-        public void Raise(int amount)
-        {
-            if (amount <= 0)
-                throw new IOException("Can't bet a negetive amount");
-            this.currentPlayer.SubstractMoney(amount);
-        }
-
-        public void AllIn()
-        {
-            this.currentPlayer.CurrentState = Player.State.ActiveAllIn;
-        }
```

Comment:
AllIn and Fold methods are still there because they change the player's state.
all the other methods (call, bet, raise) did the same thing, which is subtract money, so we don't need them (we have the SubtractMoney method).
and Check method does nothing at all so I deleted it